### DISCUSSION TITLE
Find all usages of custom attribute

### DISCUSSION THREAD

**Discussion by WybeBaT** [symfony/symfony#45134] [Q&A]
I've made a custom method Attribute. I would like to create a list of all methods where this is used, similar to debug:routes to use in a my own debug command. These Attributes will only occur in Controller methods. What is a good way to create such a list?

**Comment by ruudk** [comment]
Use this: https://github.com/symfony/symfony/pull/42039

**Reply by WybeBaT** [reply]
Thanks, I don't quite understand what's going on here. It automatically configures them? For what exactly? I'm currently reading the Attribute by running a ReflectionMethod on the method defined in the Request to process the configuration when a method is called through a route. Is that something I should replace by this somehow? And how does this help me list all occurrences of the attribute?

**Reply by ruudk** [reply]
This is how it works: https://symfony.com/blog/new-in-symfony-5-4-dependencyinjection-improvements#autoconfigurable-methods-properties-and-parameters

You add `$container->registerAttributeForAutoconfiguration` to your Extension to add tags every time the attribute is found. Then in the debug command you fetch all tags and display them.

**Reply by helmis123** [reply]
Thanks @ruudk 
Strangely this part has not been documented in symfony.

**Reply by ruudk** [reply]
It's pending documentation, see:
- https://github.com/symfony/symfony-docs/issues/15763

Maybe you can contribute some docs when you implemented it? 🙏

**Reply by WybeBaT** [reply]
> You add $container->registerAttributeForAutoconfiguration to your Extension

I'm not using an extension. Should I make on for this purpose?

**Reply by ruudk** [reply]
Yes, you can only use this in an extension, or directly in the Kernel I assume...

**Reply by WybeBaT** [reply]
> You add `$container->registerAttributeForAutoconfiguration` to your Extension to add tags every time the attribute is found. Then in the debug command you fetch all tags and display them.

Does this have any impact on performance on production if I have a lot of controllers and many methods in them? Or does this only run when you run the debug command?

**Reply by ruudk** [reply]
extensions only run during build phase of the cache, so no :)

**Reply by ruudk** [reply]
and the debug command reads the tags from cache

**Reply by WybeBaT** [reply]
Perfect, thanks!

**Reply by helmis123** [reply]
@WybeBaT So u can contribute with some docs ? 🙏

**Reply by WybeBaT** [reply]
@helmis123 Implementing this is currently on the backlog but if I get to it, I'll see if I can write some documentation for it.
